### PR TITLE
[libpostal] add build script for libpostal

### PR DESCRIPTION
**Pull request by mattwigway**
This adds a build script for [libpostal](https://github.com/openvenues/libpostal), an address-parsing library. I intend to write a Julia wrapper for it (WIP at [mattwigway/Postal.jl](https://github.com/mattwigway/Postal.jl])).

I do have two questions about how to package this:
1. libpostal doesn't really follow a versioning scheme; the last release was seven years ago and everyone seems to just use master. I'm not sure how to version the JLL.
2. I'm getting a warning about the binaries using avx2 instructions on the w64 builds. I tried adding -mno-avx2 to CFLAGS but that didn't solve the problem, I'm not sure what else to try.

### CODE REVIEW

**Comment by imciner2** [comment]
> I'm getting a warning about the binaries using avx2 instructions on the w64 builds. I tried adding -mno-avx2 to CFLAGS but that didn't solve the problem, I'm not sure what else to try.

You can safely ignore this probably. The auditor seems to always give that warning, even in cases where no problematic instructions are present.

File: L/libpostal/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --disable-data-download ${flags}
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# aarch64-linux has an assembler error: "Error: conditional branch out of range"
+# This is a rare enough platform we just exclude for now
+platforms = expand_cxxstring_abis(supported_platforms(; experimental=true, exclude=p -> p.tags["os"] == "linux" && p.tags["arch"] == "aarch64"))
```
Comment (imciner2) [inline L/libpostal/build_tarballs.jl:39]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms(; exclude=p -> p.tags["os"] == "linux" && p.tags["arch"] == "aarch64"))
```

**Comment by imciner2** [comment]
I see in the artifact that it includes `libpostal.a`, which is the static library. We generally don't like to include static libraries in the JLL packages (unless absolutely necessary) because they bloat the storage and download sizes of the packages and the end users don't ever need them. I'd suggest disabling the static library build.

**Comment by mattwigway** [comment]
@imciner2 should be fixed, thanks!

File: L/libpostal/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --disable-data-download ${flags}
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# aarch64-linux has an assembler error: "Error: conditional branch out of range"
+# This is a rare enough platform we just exclude for now
+platforms = expand_cxxstring_abis(supported_platforms(; exclude=p -> p.tags["os"] == "linux" && p.tags["arch"] == "aarch64"))
```
Comment (giordano) [inline L/libpostal/build_tarballs.jl:39]: I don't agree aarch64-linux is that rare.  Also, this seems to be https://gcc.gnu.org/legacy-ml/gcc-patches/2018-01/msg01808.html, which was fixed around GCC v8
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: L/libpostal/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    # libpostal_data not available on Windows.
+    #ExecutableProduct("libpostal_data", :libpostal_data)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (giordano) [inline L/libpostal/build_tarballs.jl:53]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
               julia_compat="1.6", preferred_gcc_version=v"8")
```

**Comment by mattwigway** [comment]
@giordano sure enough, thanks for the tip! Fixed now and the build works for aarch64-linux

**Pull request merged**